client-runtime/src/test/java/com/microsoft/rest/v2/util/ByteBufferUtilTests.java

```diff
@@ -0,0 +1,30 @@
+package com.microsoft.rest.v2.util;
+
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+import static org.junit.Assert.*;
+
+public class ByteBufferUtilTests {
+    @Test
+    public void toByteArrayWithNullByteBuffer()
+    {
+        assertNull(ByteBufferUtil.toByteArray(null));
+    }
+
+    @Test
+    public void toByteArrayWithEmptyByteBuffer()
+    {
+        assertArrayEquals(new byte[0], ByteBufferUtil.toByteArray(ByteBuffer.wrap(new byte[0])));
+    }
+
+    @Test
+    public void toByteArrayWithNonEmptyByteBuffer()
+    {
+        final ByteBuffer byteBuffer = ByteBuffer.wrap(new byte[] { 0, 1, 2, 3, 4 });
+        assertEquals(5, byteBuffer.remaining());
+        assertArrayEquals(new byte[] { 0, 1, 2, 3, 4 }, ByteBufferUtil.toByteArray(byteBuffer));
+        assertEquals(0, byteBuffer.remaining());
```

Comment:
Depending on the behavior you settle on it might be reasonable to assert that the return value *is not* the same byte[] instance as the .array() on the byteBuffer.

Reply:
Done.